cluster.tf

```diff
@@ -1,104 +1,53 @@
-  flavor            = each.value.machine_type
-  worker_count      = each.value.workers_per_subnet
-  kube_version = (
-    lookup(each.value, "kube_version", null) == "latest" # if version is latest
-    || lookup(each.value, "kube_version", null) == null  # or if version is null
-    ? local.latest_kube_version[each.value.kube_type]    # use latest
-    : each.value.kube_version                            # otherwise use value
-  )
-  update_all_workers = lookup(each.value, "update_all_workers", null)
-  tags               = var.tags
-  wait_till          = var.wait_till
-  entitlement        = each.value.entitlement
-  cos_instance_crn   = each.value.cos_instance_crn
-  pod_subnet         = each.value.pod_subnet
-  service_subnet     = each.value.service_subnet
-
-  dynamic "zones" {
-    for_each = each.value.subnets
-    content {
-      subnet_id = zones.value["id"]
-      name      = zones.value["zone"]
-    }
-  }
-
-  dynamic "kms_config" {
-    for_each = each.value.kms_config == null ? [] : [each.value.kms_config]
-    content {
-      crk_id           = module.key_management.key_map[kms_config.value.crk_name].key_id
-      instance_id      = module.key_management.key_management_guid
-      private_endpoint = kms_config.value.private_endpoint
-    }
-  }
-
-  disable_public_service_endpoint = true
-
-  timeouts {
-    create = "3h"
-    delete = "2h"
-    update = "3h"
-  }
-
-}
-
-##############################################################################
-
-
-##############################################################################
-# Create Worker Pools
-##############################################################################
-
-resource "ibm_container_vpc_worker_pool" "pool" {
-  for_each          = local.worker_pools_map
+  region            = var.region
+  cluster_name      = each.value.cluster_name
   vpc_id            = each.value.vpc_id
-  resource_group_id = local.resource_groups[each.value.resource_group]
-  entitlement       = each.value.entitlement
-  cluster           = ibm_container_vpc_cluster.cluster[each.value.cluster_name].id
-  worker_pool_name  = each.value.name
-  flavor            = each.value.flavor
-  worker_count      = each.value.workers_per_subnet
-
-  dynamic "zones" {
-    for_each = each.value.subnets
-    content {
-      subnet_id = zones.value["id"]
-      name      = zones.value["zone"]
+  vpc_subnets = {
+    vsi-zone-1 = [
+      for zone in each.value.subnets :
+      {
+        id         = zone.id
+        zone       = zone.zone
+        cidr_block = zone.cidr
+      }
+    ]
+  }
+  worker_pools = [
+    for pool in each.value.worker_pools :
+    {
+      subnet_prefix    = pool.subnet_names[0]
+      pool_name        = pool.name
+      machine_type     = pool.flavor
+      workers_per_zone = pool.workers_per_subnet
     }
+  ]
+  ocp_version             = each.value.ocp_version
+  tags                    = var.resource_tags
```

Comment:
Missing entitlement so looks to default to cloud_paks which most customers don't have

Reply:
added this parameter.